### ISSUE TITLE
Highlighting Incorrect in Babel ES2015 Code

### ISSUE THREAD

**Issue by cphoover** [gotwarlost/istanbul#557]
When running coverage against babel es2015 source code, highlighting is completely off.

It seems to start perhaps in the correct position, but the end of the highlighting is off.

Here is an annotated screenshot of the issue: 
http://imgur.com/xXFWjtp

**Comment by skozin** [comment]
I've the same issue. Highlighting is incorrect when using Babel to transpile ES2015 to ES5, and even ES5 to ES5 (i.e. running Babel over already compiled sources). Example:

![image](https://cloud.githubusercontent.com/assets/1699593/13902758/699706e2-ee68-11e5-86b7-edda718e6200.png)

![image](https://cloud.githubusercontent.com/assets/1699593/13902754/4f6da0a0-ee68-11e5-877c-7f3aef4f2fa5.png)

**Comment by cphoover** [comment]
@skozin as a temporary solution... I changed the css so that .no-branch has a left border of 10px solid red... instead of a background color... This way you only see highlighting at the start of the missed coverage.

**Comment by skozin** [comment]
@cphoover, thanks, interesting hack. I'll try it =)

**Comment by julmot** [comment]
I also advocate that Babel's workaround code will not be tested. Example:
![example](https://cloud.githubusercontent.com/assets/8310677/14170816/ee2b6b8a-f72f-11e5-9054-456152fc10a4.png)

**Comment by skozin** [comment]
@julmot, +1, but the only way for this probably is to use info from sourcemaps to see whether the specific part of the generated code maps to some location in the sources, and exclude it from report if not (i.e. don't highlight it and don't count in covered/total functions/branches/statements/lines numbers).

Probably this can be done only on function level, as Babel only generates helpers in form of functions. On the other hand, Babel uses regenerator to transpile generators and async/await functions, and regenerator produces statements that cannot be mapped to the original source.

But I would be happy even if Istanbul covered only the non-generator case.

**Comment by toddpi314** [comment]
@cphoover Agreed. This appears to just be proble with utilizing the `background` css attribute and the container structure. If you favor `border` vs `background` the correct areas are highlighted
